sympy/core/tests/test_relational.py

```diff
@@ -661,3 +661,9 @@ def test_issue_10304():
     assert d.is_comparable is False  # if this fails, find a new d
     e = 1 + d*I
     assert simplify(Eq(e, 0)) is S.false
+
+
+def test_issue_10401():
+    x = Symbol('x')
+    assert simplify(Eq(1 / (x+1), 0)) == False
+    assert simplify(Eq(x / (x+1), 1)) == False
```

Comment:
is_zero still calls the assumptions system so I'm still -0.5 to doing it. I'd like to reduce the amount of calls to the assumptions system that happens in constructors.